### DISCUSSION TITLE
How to code the drum song in .h2song

### DISCUSSION THREAD

**Discussion by vinodpkd** [hydrogen-music/hydrogen#1993] [Hydrogen on Windows]
Song :
![image](https://github.com/user-attachments/assets/ea2ace2f-8859-414e-be5b-26d2931ee417)

Drum Notation Ref
![image](https://github.com/user-attachments/assets/a2e28b46-0828-45ec-96ed-49994f901ebd)

bpm = 80
How to code this song as .h2song xml file.( I am seeing GM_kit_demo1.h2song)

```
<note>
     <position>??</position>
     <leadlag>0</leadlag>
     <velocity>0.22</velocity>
     <pan>0</pan>
     <pitch>0</pitch>
     <probability>1</probability>
     <key>C0</key>
     <length>-1</length>
     <instrument>??</instrument>
     <note_off>false</note_off>
    </note>

```
How to fill this position tag and instrument tag wrt timing of each instrument.

**Comment by theGreatWhiteShark** [comment]
Hey @vinodpkd,

Hmm. Well, it is best to use the [Pattern Editor](http://hydrogen-music.org/documentation/manual/manual_en.html#chpt.pattern_editor) and try different placements until the result sound the same as in the video. When everything is done, you can [save the song](http://hydrogen-music.org/documentation/manual/manual_en.html#sect.main_menu.projects) and the resulting XML file will have proper position and instrument data.

If you need to do this in a XML file directly:
- `position` is given in **ticks** relative to the start of the pattern. Ticks are the smallest temporal unit that can be resolved for notes in Hydrogen. The current tick size is `sample rate * 60 / tempo in BPM / song resolution`.
- `instrument` corresponds to the `id` of an instrument in the `drumkit.xml` definition of the kit you are using when creating the pattern.
